MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/navigation/TwoHandNavigator.kt

```diff
@@ -0,0 +1,163 @@
+        }
+        if(swapHands) {
+            tempNodesList.swap(firstHandIndex, firstHandIndex + 1)
+        }
+        nodes = tempNodesList
+    }
+
+    override fun node(identifier: String): Node? {
+        return nodes.find { it.identifier == identifier }
+    }
+
+    override fun hasNodeAfter(currentNode: Node, branchResult: BranchNodeResult): Boolean {
+        return nextNode(currentNode.identifier, currentHandSelection(branchResult)) != null
+    }
+
+    override fun nodeAfter(currentNode: Node?, branchResult: BranchNodeResult): NavigationPoint {
+        return NavigationPoint(
+            nextNode(currentNode?.identifier, currentHandSelection(branchResult)),
+            branchResult,
+            Direction.Forward)
+    }
+
+    override fun nodeBefore(currentNode: Node?, branchResult: BranchNodeResult): NavigationPoint {
+        return NavigationPoint(previousNode(currentNode), branchResult, Direction.Backward)
+    }
+
+    override fun allowBackNavigation(currentNode: Node, branchResult: BranchNodeResult): Boolean {
+        return previousNode(currentNode) != null
+    }
+
+    override fun progress(currentNode: Node, branchResult: BranchNodeResult): Progress? {
+        return null
+    }
+
+    override fun isCompleted(currentNode: Node, branchResult: BranchNodeResult): Boolean {
+        return isCompleted(currentNode)
+    }
+
+    private fun currentHandSelection(branchResult: BranchNodeResult): HandSelection? {
+        val answer = branchResult.pathHistoryResults
+            .find { it.identifier == handSelectionIdentifier } as? AnswerResult ?: return null
+        val rawValue = answer.jsonValue.toString().replace("\"", "").uppercase()
+        return try {
+            HandSelection.valueOf(rawValue)
+        }
+        catch (exception: Exception) {
+            null
+        }
+    }
+
+    private fun firstNode(): Node {
+        return nodes.first()
+    }
+
+    private fun nextNode(identifier: String?, handSelection: HandSelection?): Node? {
+        if(identifier == null) {
+            return firstNode()
+        }
+        val currentNodeIndex = nodes.indexOfFirst { it.identifier == identifier }
+        if(currentNodeIndex + 1 >= nodes.size) {
+            return null
+        }
+        val nextNode = nodes[currentNodeIndex + 1]
+        if(handSelection == null || nextNode.hand() == null || handSelection == nextNode.hand()) {
+            return nextNode
+        }
+        if(currentNodeIndex + 2 >= nodes.size) {
+            return null
+        }
+        return nodes[currentNodeIndex + 2]
+    }
+
+    private fun previousNode(currentNode: Node?): Node? {
+        if(currentNode == null || isCompleted(currentNode)) {
+            return null
+        }
+        val currentNodeIndex = nodes.indexOfFirst { it.identifier == currentNode.identifier }
+        if(currentNodeIndex < 1) {
+            return null
+        }
+        return nodes[currentNodeIndex - 1]
```

Comment:
Needs to account for navigating back on a section step and skipping hands we don't want to navigate back to.